src/renderer/library/components/settings/Settings.tsx

```diff
@@ -235,122 +237,175 @@ const SaveCreatorSettings: React.FC<{}> = () => {
                                 className="R2_CSS_CLASS__FORCE_NO_FOCUS_OUTLINE"
                                 title={k.provider}
-                                value={k.key}
-                                onChange={(e) => {
-                                    const v = e.target.value;
-                                    updateKey(index, k.provider, v, false);
-                                }}
+                                value={newKey}
+                                onChange={(e) => setNewKey(e.target.value)}
                                 required
                             />
                             <label htmlFor="api-key">{__("settings.apiKey.keyLabel")}</label>
-                            <button type="button" onClick={(e) => {e.preventDefault(); updateInputType(index, inputType[index] === "password" ? "text" : "password");}}>
-                                <SVG ariaHidden svg={inputType[index] === "password" ? EyeOpenIcon : EyeClosedIcon} style={{width: "20px"}} />
+                            <button
+                                type="button"
+                                onClick={(e) => {
+                                    e.preventDefault();
+                                    setInputType(inputType === "password" ? "text" : "password");
+                                }}
+                            >
+                                <SVG ariaHidden svg={inputType === "password" ? EyeOpenIcon : EyeClosedIcon} style={{ width: "20px" }} />
                             </button>
                         </div>
-                        <button type="submit" title={__("settings.apiKey.validate")} disabled={k.key.length === 0 || k.provider.length === 0} onClick={(e) => {
-                                    e.preventDefault();
-                                    updateKey(index, k.provider, k.key, true);
-                                }}><SVG ariaHidden svg={k.submitted ? CheckIcon: FloppyDiskIcon} style={{width: "inherit"}} /></button>
+                        <button
+                            type="submit"
+                            title={__("settings.apiKey.validate")}
+                            disabled={!newKey || !newProvider}
+                            onClick={(e) => {
+                                e.preventDefault();
+                                updateKey();
+                            }}
+                        >
+                            <SVG ariaHidden svg={k.submitted ? CheckIcon : FloppyDiskIcon} style={{ width: "inherit" }} />
+                        </button>
                     </div>
-                    </form>
-                    }
-                </div>
+                </form>
+            )}
+        </div>
+    );
+};
+
+const ApiKeysList = () => {
+    const [__] = useTranslator();
+
+    const apiKeys = useSelector((state: ICommonRootState) => state.apiKeys);
+    const [keys, setKeys] = React.useState(apiKeys);
```

Comment:
Why do you need to store the reference to the array keys in the useState ? 

`addNewKey` is used only once when they are one or zero key stored, but why ? 
Then, why do you need to save the entire reference of the apiKeys from the redux store to append an empty apiKey object. You can just use a local boolean to show a non initialised ApiKeyComponent when the button "add new key" is clicked. 

What do you think, you overcomplicated the component I guess.